src/realm/obj.cpp

```diff
@@ -1999,6 +2052,21 @@ Obj& Obj::set_collection(ColKey col_key, CollectionType type)
     return *this;
 }
 
+Obj& Obj::set_collection(StringData prop_name, CollectionType type)
```

Comment:
Maybe we could have a similar method that only works for additional properties, so it skips checking the schema

Reply:
`set_additional_collection` has been added